ext/OceananigansReactantExt/ShardedGrids.jl

```diff
@@ -1,16 +1,128 @@
+                               z = nothing,
+                               radius = R_Earth,
+                               topology = nothing,
+                               halo = nothing)
+
+    topology, size, halo, latitude, longitude, z, precompute_metrics =
+        validate_lat_lon_grid_args(topology, size, halo, FT, latitude, longitude, z, precompute_metrics)
+
+    Nλ, Nφ, Nz = size
+    Hλ, Hφ, Hz = halo
+    TX, TY, TZ = topology
+
+    Lλ, λᶠᵃᵃ, λᶜᵃᵃ, Δλᶠᵃᵃ, Δλᶜᵃᵃ = generate_coordinate(FT, topology, size, halo, longitude, :longitude, 1, architecture)
+    Lφ, φᵃᶠᵃ, φᵃᶜᵃ, Δφᵃᶠᵃ, Δφᵃᶜᵃ = generate_coordinate(FT, topology, size, halo, latitude,  :latitude,  2, architecture)
+    Lz, z                        = generate_coordinate(FT, topology, size, halo, z,         :z,         3, architecture)
+
+    # Extracting the local range
+    xsharding  = Sharding.DimsSharding(arch.connectivity, (1,  ), (:x,   )) # X Stencil sharding
+    ysharding  = Sharding.DimsSharding(arch.connectivity, (2,  ), (:y,   )) # Y Stencil sharding
+    xysharding = Sharding.DimsSharding(arch.connectivity, (1, 2), (:x, :y)) # XY Pencil sharding
+
+    # λ and φ metrics are either 1D or a number
+    λmetric_sharding = ndims(Δλᶜᵃᵃ) == 1 ? xsharding : Reactant.Sharding.Sharding.NoSharding() # Will this work?
+    φmetric_sharding = ndims(Δφᵃᶜᵃ) == 1 ? ysharding : Reactant.Sharding.Sharding.NoSharding() # Will this work?
+
+    preliminary_grid = LatitudeLongitudeGrid{TX, TY, TZ}(architecture,
+                                                         Nλ, Nφ, Nz,
+                                                         Hλ, Hφ, Hz,
+                                                         Lλ, Lφ, Lz,
+                                                         Reactant.to_rarray(Δλᶠᵃᵃ; λmetric_sharding), 
+                                                         Reactant.to_rarray(Δλᶜᵃᵃ; λmetric_sharding), 
+                                                         Reactant.to_rarray(λᶠᵃᵃ ; xsharding), 
+                                                         Reactant.to_rarray(λᶜᵃᵃ ; xsharding),
+                                                         Reactant.to_rarray(Δφᵃᶠᵃ; φmetric_sharding), 
+                                                         Reactant.to_rarray(Δφᵃᶜᵃ; φmetric_sharding), 
+                                                         Reactant.to_rarray(φᵃᶠᵃ ; ysharding), 
+                                                         Reactant.to_rarray(φᵃᶜᵃ ; ysharding),
+                                                         Reactant.to_rarray(z), # Intentionally not sharded
+                                                         (nothing for i=1:10)..., FT(radius))
+
+    if !precompute_metrics
+        return preliminary_grid
+    end
+       
+    # Note! This step requires a kernel that launches on a `ReactantState` architecture.
+    # Would there be issues?
+    grid = with_precomputed_metrics(preliminary_grid) 
```

Comment:
```suggestion
    grid = @jit with_precomputed_metrics(preliminary_grid) 
```

not sure if this would help